lib/resolve-value.js

```diff
@@ -41,12 +39,19 @@ var resolveValue = function(decl, map, /*optional*/ignorePseudoScope, /*internal
 
 	// Resolve any var(...) substitutons
 	var isResultantValueUndefined = false;
-	resultantValue = resultantValue.replace(new RegExp(RE_VAR_FUNC.source, 'g'), function(match, variableName, fallback) {
-		// Loop through the list of declarations for that value and find the one that best matches
-		// By best match, we mean, the variable actually applies. Criteria:
-		//		- is under the same scope
-		//		- The latest defined `!important` if any
-		var matchingVarDeclMapItem;
+	
+	// var() = var( <custom-property-name> [, <any-value> ]? )
+	// matches `name[, fallback]`, captures "name" and "fallback"
+	// See: http://dev.w3.org/csswg/css-variables/#funcdef-var
+	var match;
+	while (match = balanced('var(', ')', resultantValue)) {
```

Comment:
> Seems like we should also use this same looping strategy to populate variablesUsedInValueMap just above
> 
> https://github.com/MadLittleMods/postcss-css-variables/pull/97#discussion_r279523067

We could also move `RE_VAR_FUNC` to `index.js` then

Reply:
I think it would be good to have a modified version of `RE_VAR_FUNC`, just to check if str have `var(--somethig)` declaration. Remove it would generate an extra work, refactoring `index.js`. I'm not sure if best option for a small fix to match nested css functions.